azurerm/internal/services/network/firewall_policy_rule_collection_group_resource.go

```diff
@@ -0,0 +1,860 @@
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("checking for existing Firewall Policy Rule Collection Group %q (Resource Group %q / Policy %q): %+v", name, policyId.ResourceGroup, policyId.Name, err)
+			}
+		}
+
+		if resp.ID != nil && *resp.ID != "" {
+			return tf.ImportAsExistsError("azurerm_firewall_policy_rule_collection_group", *resp.ID)
+		}
+	}
+
+	locks.ByName(policyId.Name, azureFirewallPolicyResourceName)
+	defer locks.UnlockByName(policyId.Name, azureFirewallPolicyResourceName)
+
+	param := network.FirewallPolicyRuleCollectionGroup{
+		FirewallPolicyRuleCollectionGroupProperties: &network.FirewallPolicyRuleCollectionGroupProperties{
+			Priority: utils.Int32(int32(d.Get("priority").(int))),
+		},
+	}
+	var rulesCollections []network.BasicFirewallPolicyRuleCollection
+	rulesCollections = append(rulesCollections, expandAzureRmFirewallPolicyRuleCollectionApplication(d.Get("application_rule_collection").(*schema.Set).List())...)
+	rulesCollections = append(rulesCollections, expandAzureRmFirewallPolicyRuleCollectionNetwork(d.Get("network_rule_collection").(*schema.Set).List())...)
+	rulesCollections = append(rulesCollections, expandAzureRmFirewallPolicyRuleCollectionNat(d.Get("nat_rule_collection").(*schema.Set).List())...)
+	param.FirewallPolicyRuleCollectionGroupProperties.RuleCollections = &rulesCollections
+
+	future, err := client.CreateOrUpdate(ctx, policyId.ResourceGroup, policyId.Name, name, param)
+	if err != nil {
+		return fmt.Errorf("creating Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q): %+v", name, policyId.ResourceGroup, policyId.Name, err)
+	}
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q): %+v", name, policyId.ResourceGroup, policyId.Name, err)
+	}
+
+	resp, err := client.Get(ctx, policyId.ResourceGroup, policyId.Name, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q): %+v", name, policyId.ResourceGroup, policyId.Name, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q) ID", name, policyId.ResourceGroup, policyId.Name)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmFirewallPolicyRuleCollectionGroupRead(d, meta)
+}
+
+func resourceArmFirewallPolicyRuleCollectionGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.FirewallPolicyRuleGroupClient
+	policyClient := meta.(*clients.Client).Network.FirewallPolicyClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FirewallPolicyRuleCollectionGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	policyResp, err := policyClient.Get(ctx, id.ResourceGroup, id.PolicyName, "")
```

Comment:
Why do we need to check the existence of the linked policy before retrieving the rule collection group?

Reply:
The reason is that we will need to use the `policyResp` to set the `firewall_policy_id` later. Instead, we should opt in to using the ID Formatter to save this API call.